### ISSUE TITLE
migrations before gitaly start won't work on fresh install

### ISSUE THREAD

**Issue by fpeyraud** [sameersbn/docker-gitlab#1617]
Hi,
I've tried to use the proposed docker-compose.yml to start a fresh install and it did not work.
The startup process stopped with a message "Missing Rails.application.secrets.openid_connect_signing_key for production environment. The secret will be generated and stored in config/secrets.yml.".
I removed the `> /dev/null` in the following call : 
```
exec_as_git force=yes bundle exec rake gitlab:setup \
      ${GITLAB_ROOT_PASSWORD:+GITLAB_ROOT_PASSWORD=$GITLAB_ROOT_PASSWORD} \
      ${GITLAB_ROOT_EMAIL:+GITLAB_ROOT_EMAIL=$GITLAB_ROOT_EMAIL} >/dev/null
```
([look in the code](https://github.com/sameersbn/docker-gitlab/blob/8b638b37adb2c15fc6af8204c52c6015c2eb484b/assets/runtime/functions#L1638)) 
and relaunched the process, and the error popped out : 
`Failed to connect to Gitaly...
Error: 14:Connect Failed
`
Starting supervisord, which automatically starts gitaly allowed to got further in the migration process. I think supervisor should be started before calling `migrate_database`

In order to reproduce here is my docker-compose.yml
```version: '2'

services:
  redis:
    restart: always
    image: sameersbn/redis:latest
    command:
    - --loglevel warning
    volumes:
    - /srv/data/gitlab/redis:/var/lib/redis:Z
    networks:
    - int
    logging:
      driver: syslog
      options:
        syslog-address: "unixgram:///dev/log"
        tag: gitlab-prod-redis


  postgresql:
    restart: always
    image: sameersbn/postgresql:9.6-2
    volumes:
    - /srv/data/gitlab/postgresql:/var/lib/postgresql:Z
    environment:
    - DB_USER=gitlab
    - DB_PASS=password
    - DB_NAME=gitlabhq_production
    - DB_EXTENSION=pg_trgm
    networks:
    - int
    logging:
      driver: syslog
      options:
        syslog-address: "unixgram:///dev/log"
        tag: gitlab-prod-pg

  gitlab:
    restart: always
    image: sameersbn/gitlab:10.8.1
    depends_on:
    - redis
    - postgresql
    ports:
    - "10022:22"
    volumes:
    - /srv/data/gitlab/gitlab:/home/git/data:Z
    networks:
    - int
    - webgw
    logging:
      driver: syslog
      options:
        syslog-address: "unixgram:///dev/log"
        tag: gitlab-prod-app
    labels:
      - "traefik.backend=gitlab-prod-frontdev"
      - "traefik.frontend.rule=Host:gitlab-prod.mywebsite.io"
      - "traefik.frontend.entryPoints=https"
      - "traefik.docker.network=webgateway"
      - "traefik.enable=true"
    environment:
    - DEBUG=true

    - DB_ADAPTER=postgresql
    - DB_HOST=postgresql
    - DB_PORT=5432
    - DB_USER=gitlab
    - DB_PASS=password
    - DB_NAME=gitlabhq_production

    - REDIS_HOST=redis
    - REDIS_PORT=6379

    - TZ=Europe/Paris
    - GITLAB_TIMEZONE=Paris

    - GITLAB_HTTPS=false
    - SSL_SELF_SIGNED=false

    - GITLAB_HOST=localhost
    - GITLAB_PORT=10080
    - GITLAB_SSH_PORT=10022
    - GITLAB_RELATIVE_URL_ROOT=
    - GITLAB_SECRETS_DB_KEY_BASE=YmIwYTZNzYwMTY5YjQzZGQxOTg2ZTMzZWRkZTkw
    - GITLAB_SECRETS_SECRET_KEY_BASE=MTk3ZTI4MDjUxNTZjM0M2E4YTZiMTlkZWI0OWNlODZh
    - GITLAB_SECRETS_OTP_KEY_BASE=M2ZkNGIwGRiNDI4MGU0OTgyYzZlOWM5OWFkNWE2Mjdh

    - GITLAB_ROOT_PASSWORD=xF3AMtG5Qh
    - GITLAB_ROOT_EMAIL=infra+gitlab@example.com

    - GITLAB_NOTIFY_ON_BROKEN_BUILDS=true
    - GITLAB_NOTIFY_PUSHER=false

    - GITLAB_EMAIL=gitlab@example.com
    - GITLAB_EMAIL_REPLY_TO=noreply@example.com
    - GITLAB_INCOMING_EMAIL_ADDRESS=gitlab@example

    - GITLAB_BACKUP_SCHEDULE=daily
    - GITLAB_BACKUP_TIME=01:00

    - SMTP_ENABLED=false
    - SMTP_DOMAIN=example.com
    - SMTP_HOST=mail.example.priv
    - SMTP_PORT=25
    #- SMTP_USER=mailer@example.com
    #- SMTP_PASS=password
    #- SMTP_STARTTLS=true
    #- SMTP_AUTHENTICATION=login

    - IMAP_ENABLED=false
    - IMAP_HOST=imap.gmail.com
    - IMAP_PORT=993
    - IMAP_USER=mailer@example.com
    - IMAP_PASS=password
    - IMAP_SSL=true
    - IMAP_STARTTLS=false

    - OAUTH_ENABLED=false
    - OAUTH_AUTO_SIGN_IN_WITH_PROVIDER=
    - OAUTH_ALLOW_SSO=
    - OAUTH_BLOCK_AUTO_CREATED_USERS=true
    - OAUTH_AUTO_LINK_LDAP_USER=false
    - OAUTH_AUTO_LINK_SAML_USER=false
    - OAUTH_EXTERNAL_PROVIDERS=

    - OAUTH_CAS3_LABEL=cas3
    - OAUTH_CAS3_SERVER=
    - OAUTH_CAS3_DISABLE_SSL_VERIFICATION=false
    - OAUTH_CAS3_LOGIN_URL=/cas/login
    - OAUTH_CAS3_VALIDATE_URL=/cas/p3/serviceValidate
    - OAUTH_CAS3_LOGOUT_URL=/cas/logout

    - OAUTH_GOOGLE_API_KEY=
    - OAUTH_GOOGLE_APP_SECRET=
    - OAUTH_GOOGLE_RESTRICT_DOMAIN=

    - OAUTH_FACEBOOK_API_KEY=
    - OAUTH_FACEBOOK_APP_SECRET=

    - OAUTH_TWITTER_API_KEY=
    - OAUTH_TWITTER_APP_SECRET=

    - OAUTH_GITHUB_API_KEY=
    - OAUTH_GITHUB_APP_SECRET=
    - OAUTH_GITHUB_URL=
    - OAUTH_GITHUB_VERIFY_SSL=

    - OAUTH_GITLAB_API_KEY=
    - OAUTH_GITLAB_APP_SECRET=

    - OAUTH_BITBUCKET_API_KEY=
    - OAUTH_BITBUCKET_APP_SECRET=

    - OAUTH_SAML_ASSERTION_CONSUMER_SERVICE_URL=
    - OAUTH_SAML_IDP_CERT_FINGERPRINT=
    - OAUTH_SAML_IDP_SSO_TARGET_URL=
    - OAUTH_SAML_ISSUER=
    - OAUTH_SAML_LABEL="Our SAML Provider"
    - OAUTH_SAML_NAME_IDENTIFIER_FORMAT=urn:oasis:names:tc:SAML:2.0:nameid-format:transient
    - OAUTH_SAML_GROUPS_ATTRIBUTE=
    - OAUTH_SAML_EXTERNAL_GROUPS=
    - OAUTH_SAML_ATTRIBUTE_STATEMENTS_EMAIL=
    - OAUTH_SAML_ATTRIBUTE_STATEMENTS_NAME=
    - OAUTH_SAML_ATTRIBUTE_STATEMENTS_FIRST_NAME=
    - OAUTH_SAML_ATTRIBUTE_STATEMENTS_LAST_NAME=

    - OAUTH_CROWD_SERVER_URL=
    - OAUTH_CROWD_APP_NAME=
    - OAUTH_CROWD_APP_PASSWORD=

    - OAUTH_AUTH0_CLIENT_ID=
    - OAUTH_AUTH0_CLIENT_SECRET=
    - OAUTH_AUTH0_DOMAIN=

    - OAUTH_AZURE_API_KEY=
    - OAUTH_AZURE_API_SECRET=
    - OAUTH_AZURE_TENANT_ID=
networks:
    webgw:
        external:
            name: webgateway
    int:
        driver: bridge
```

**Comment by solidnerd** [comment]
Hey @fpeyraud,

thanks for setting up this issue. Something went wrong on my e2e tests. I can confirm this issue now. I need to think about it how we can solve it in an elegant way.

**Comment by solidnerd** [comment]
Hey, 

it will be fixed with `10.8.3`. If it's not the case for you please reopen this issue.
